### PR TITLE
Android embedding v2

### PR DESCRIPTION
**Pull request by ikurek**
This PR is a migration supposed to solve #311, by fully adapting Flutter Android Embedding V2 following the [official migration guide](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects). Besides the migration itself, additional compatibility changes include:
* Forced Kotlin version for example app (Flutter recommends setting 1.5.31)
* Changed Android target and compile SDK versions to latest (32)
* Updated a few Flutter dependencies for example app
* Updated test dependency version for dart library
* Set Android application as exported in Manifest file, to comply with Android S requirements

### CODE REVIEW

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -41,7 +41,7 @@ android {
     defaultConfig {
         applicationId "io.ably.flutter.example"
         minSdkVersion 19
-        targetSdkVersion 30
+        targetSdkVersion 32
```
Comment (KacperKluka) [inline example/android/app/build.gradle:44]: Is there a particular reason why you've chosen `32` and not `31`? :thinking: I think that version `32` isn't released yet or is currently in beta, right? Here's a list of the versions https://developer.android.com/studio/releases/platforms :wink:

**Review by KacperKluka — COMMENTED** [review]
Looks good but I'm not sure about one thing :grin:

File: example/android/app/build.gradle (hunk 1 above)
Comment (ikurek) [reply example/android/app/build.gradle:44]: Android Studio suggest version 32, when it's set to 31. I'm not using any Preview/Canary version, so I thought that the suggested one is the latest stable 🤔 I'll downgrade to 31 then 🙂

Zrzut ekranu 2022-02-4 o 14 11 58

Comment (KacperKluka) [reply example/android/app/build.gradle]: That's strange :sweat_smile: I've checked and in the AAT project Android Studio suggests me version 31. I wonder why :thinking: 
image

Comment (ikurek) [reply example/android/app/build.gradle]: @KacperKluka downgraded to SDK 31 in https://github.com/ably/ably-flutter/pull/312/commits/c198df56bf5e142d9592284818fd53e5e895fa9c

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Review by QuintinWillison — APPROVED** [review]
Thanks. I'm keen to get this landed, in order to re-achieve green CI in the `main` branch. 🚀

**Pull request merged**